### PR TITLE
Reolink presets

### PR DESCRIPTION
**Pull request by apocaliss92**

### CODE REVIEW

File: plugins/reolink/src/main.ts [hunk 1]
```diff
@@ -134,9 +163,17 @@ class ReolinkCamera extends RtspSmartCamera implements Camera, DeviceProvider, R
...
+            const choices = this.storageSettings.values.cachedPresets.map((preset) => preset.id + '=' + preset.name);
+            return {
+                choices,
+            };
+        };
+
         this.updateDeviceInfo();
         (async () => {
-            this.updatePtzCaps();
+            await this.updatePtzCaps();
```
Comment (koush) [inline plugins/reolink/src/main.ts:176]: this call may fail thats why it does not await.

File: plugins/reolink/src/reolink-api.ts [hunk 2]
```diff
@@ -213,45 +218,56 @@ export class ReolinkCameraClient {
...
+    private async ptzOp(op: string, speed?: number, id?: string) {
+        const url = new URL(`http://${this.host}/api.cgi`);
+        const params = url.searchParams;
+        params.set('cmd', 'PtzCtrl');
+        const body = [
             {
-                cmd: "PtzCtrl",
-                param: {
-                    channel: this.channelId,
+                "cmd": "PtzCtrl",
```
Comment (koush) [inline plugins/reolink/src/reolink-api.ts:244]: remove the style changes (quotes) for a cleaner diff.

File: plugins/reolink/src/reolink-api.ts [hunk 3]
```diff
@@ -213,45 +218,56 @@ export class ReolinkCameraClient {
...
                     speed,
-                    timeout: 1,
+                    id: id ? Number(id) : undefined
                 }
             },
-        ]));
+        ];
 
-        await sleep(500);
+        const createReadable = (data: any) => {
```
Comment (koush) [inline plugins/reolink/src/reolink-api.ts:254]: make this a common function rather than copy paste

File: plugins/reolink/src/reolink-api.ts [hunk 4]
```diff
@@ -213,45 +218,56 @@ export class ReolinkCameraClient {
...
-        ]));
-
-        this.console.log(await c1);
-        this.console.log(await c2);
+        }, createReadable(body));
+        this.console.log(`Response for PtzCtrl is ${JSON.stringify(response)}`);
     }
 
     async ptz(command: PanTiltZoomCommand) {
-        // reolink doesnt accept signed values to ptz
```
Comment (koush) [inline plugins/reolink/src/reolink-api.ts:252]: why are these comments removed?

**Comment by koush** [comment]
this change has a lot of unrelated changes in it and its hard to follow. please provide a minimal change with support for only presets.

File: plugins/reolink/src/reolink-api.ts [hunk 5]
```diff
@@ -213,45 +218,56 @@ export class ReolinkCameraClient {
...
                     op,
                     speed,
-                    timeout: 1,
+                    id: id ? Number(id) : undefined
                 }
             },
-        ]));
+        ];
 
-        await sleep(500);
```
[the remark this answers is not in the document]
Comment (koush) [reply plugins/reolink/src/reolink-api.ts:232]: im not sure why this change is necessary for presets

[the remark this answers is not in the document]
Comment (apocaliss92) [reply plugins/reolink/src/reolink-api.ts:232]: It's moved in another block but I will remove all the changes not related and open another pr to fix the tilt issue
